src/prefect/serializers.py

```diff
@@ -91,6 +142,12 @@ def loads(self, blob: bytes) -> D:
     class Config:
         extra = "forbid"
 
+    @classmethod
+    def __dispatch_key__(cls):
+        if explicit_type := cls.__fields__.get("type"):
+            return explicit_type.get_default()
```

Comment:
unfortunately, because users can / have implemented their own `Serializer` subclasses where we have previously required the specification of a `type` due to `add_type_dispatch`, we will need to respect this to avoid breaking custom serializers